ingestion/src/metadata/utils/connections.py

```diff
@@ -518,3 +522,20 @@ def _(connection: TableauClient) -> None:
         raise SourceConnectionException(
             f"Unknown error connecting with {connection} - {err}."
         )
+
+
+@get_connection.register
+def _(connection: PowerBIConnection, verbose: bool = False):
+    from metadata.utils.powerbi_client import PowerBiApiClient
+
+    return PowerBiClient(PowerBiApiClient(connection))
+
+
+@test_connection.register
+def _(connection: PowerBiClient) -> None:
+    try:
+        connection.client.fetch_dashboards()
```

Comment:
is there any other method we can test here that is a bit more lightweight?

Reply:
Let me see if there is any other lightweight API we can use

Reply:
I checked other APIs but they are also similar to this one. These are the list of [supported APIs](https://docs.microsoft.com/en-us/power-bi/enterprise/read-only-apis-service-principal-authentication)

Reply:
oh I see, then let's keep this one for now. Thanks